### PR TITLE
feat: Check the package manager

### PR DESCRIPTION
**Pull request by paxperscientiam**
So, this PR begins to address #1690. Checks if user is using yarn, then pnpm, and then falls to npm. What do you think about this heuristic -- too simple?

I noticed that a Sparky task uses npm. Would you want me to address that in this PR or another one?

### CODE REVIEW

**Comment by SephReed** [comment]
I don't know yarn (or know if you know it).  Here's a quick link I found on syntax differences between it and npm:

https://stackoverflow.com/a/40137938/4808079

**Comment by paxperscientiam** [comment]
@SephReed thanks dude. I've used yarn lots, but was a bit hasty. I'll fix

File: src/env.ts [hunk 1]
```diff
@@ -19,6 +20,24 @@ export const env = {
...
+  }
+
+  if (fs.existsSync(path.join(FUSE_ROOT, './.yarnrc'))
+      || fs.existsSync(path.join(FUSE_ROOT, './yarn.lock'))) {
+    return new Data('yarn', 'yarn add', 'yarn add --dev');
+  } else if (fs.existsSync(path.join(FUSE_ROOT, './pnpm-lock.yaml'))) {
+    return new Data('pnpm', 'pnpm add', 'pnpm add --save-dev');
+  } else {
+    // package-lock.json
+    return new Data('npm', 'npm install', 'npm install --save-dev');;
```
Comment (nchanged) [inline src/env.ts]: can't you use just return `{name : "npm", ...}`
Overall it looks good, but I don't like this class here ;-)

Comment (paxperscientiam) [reply src/env.ts]: fixed

File: src/env.ts [hunk 2]
```diff
@@ -21,20 +21,16 @@ export const env = {
 };
 
 export function getPackageManagerData() {
-  function Data(name: string, installCmd: string, installDevCmd: string) {
-    this.name = name;
-    this.installCmd = installCmd;
-    this.installDevCmd = installDevCmd;
-  }
-
+  console.log("SHIT");
```
Comment (SephReed) [inline src/env.ts]: Heh.  Maybe take it out though.

**Comment by paxperscientiam** [comment]
@nchanged Not really sure what to do with conflicting package-lock.json and package.json

File: src/env.ts (hunk 2 above)
Comment (paxperscientiam) [reply src/env.ts]: done lol

**Comment by nchanged** [comment]
@paxperscientiam could you re-open the PR againts `grand-refactor` branch?

**Pull request closed without merging**